### PR TITLE
Support passing ui:label = false for Material UI

### PR DESCRIPTION
**Pull request by stanlemon**
### Reasons for making this change

[Please describe them here]

If this is related to existing tickets, include links to them as well.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by epicfaace** [comment]
I guess this approach (passing `displayLabel` to widgets) is required because in the material-ui theme, the label is handled from the widgets themselves (rather than in the [FieldTemplate directly as in the core bootstrap theme](https://github.com/rjsf-team/react-jsonschema-form/blob/49321786f79e305e16b8887d1ecd88149e250ddf/packages/core/src/components/fields/SchemaField.js#L140)).

Any thoughts on how we could clarify the documentation to note this fact (that `displayLabel` for widgets in bootstrap is not really useful)?

**Comment by stanlemon** [comment]
@epicfaace I've actually been wondering if maybe this code... https://github.com/rjsf-team/react-jsonschema-form/blob/49321786f79e305e16b8887d1ecd88149e250ddf/packages/core/src/components/fields/SchemaField.js#L265-L279 should be extracted into `utils` so it could be used on `uiSchema` from any template. That would make this a little cleaner in my opinion. You can set `label = false` in any of the Material UI widgets to get the same effect as in the Bootstrap theme, it's just as you note, the location of where this decision has to be made is fundamentally different in the Material UI theme.

**Comment by stanlemon** [comment]
@epicfaace I tried something totally different, which I think I prefer. What do you think?

**Comment by stanlemon** [comment]
@epicfaace just checking in on this one to see if you had any thoughts on proceeding with the latest approach?

File: packages/core/index.d.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ declare module '@rjsf/core' {
         > {
         id: string;
         schema: JSONSchema7;
+        uiSchema: UiSchema;
```
Comment (epicfaace) [inline packages/core/index.d.ts:102]: Can you update documentation to include uiSchema as a prop passed to all widgets?

Comment (stanlemon) [reply packages/core/index.d.ts:102]: @epicfaace just pushed, I think I added it to the right place. 🤞

File: packages/core/src/utils.js [hunk 2]
```diff
@@ -368,6 +368,26 @@ export function getUiOptions(uiSchema) {
     }, {});
 }
 
+export function getDisplayLabel(schema, uiSchema, rootSchema) {
```
Comment (stanlemon) [inline packages/core/src/utils.js:371]: @epicfaace something I thought of this morning... even though I suspect this is covered anecdotally in base tests, should I add a specific unit test for the new extracted method?

File: docs/advanced-customization/custom-widgets-fields.md [hunk 3]
```diff
@@ -121,6 +121,7 @@ The following props are passed to custom widget components:
 
 - `id`: The generated id for this field;
 - `schema`: The JSONSchema subschema object for this field;
+- `uiSchema`: The [uiSchema](#the-uischema-object) for this field;
```
Comment (epicfaace) [inline docs/advanced-customization/custom-widgets-fields.md:124]: I think this is a dead link.

File: packages/core/src/utils.js (hunk 2 above)
Comment (epicfaace) [reply packages/core/src/utils.js:371]: Yeah, that would also be good, especially because it is now exported in utils.

Comment (stanlemon) [reply packages/core/src/utils.js:371]: Check out the tests I added and let me know if you think I should add any others.

Comment (epicfaace) [reply packages/core/src/utils.js:371]: Looks good!

**Pull request merged**